lib/_http_incoming.js

```diff
@@ -349,6 +361,17 @@ IncomingMessage.prototype._dump = function _dump() {
   }
 };
 
+function onError(instance, cb, error) {
+  // This is to keep backward compatible behavior.
+  // An error is emitted only if there are listeners attached to
+  // the event.
+  if (instance.listenerCount('error') > 0) {
+    cb(error);
+  } else {
+    cb();
+  }
```

Comment:
IMO this condition is too permissive. In the previous implementation, IncomingMessage doesn't propagate only abort errors if error listener count === 0. After latest changes all errors don't propagated if error listener count === 0.
```suggestion
  if (instance.listenerCount('error') === 0 && error.code === 'ECONNRESET' && error.message === 'aborted') {
    cb();
  } else {
    cb(error);
  }
```

Reply:
Good catch, thanks, @wa-Nadoo . I intended to keep the same behavior, so I should implement the stricter check that you suggested.

Reply:
@wa-Nadoo is correct.

Reply:
`IncomingMessage.destroy(new Error())` should be unhandled exception

Reply:
I think I got confused, the check is indeed in functions that should be called only when aborting if I am not mistaken. I'll fix this and add a test.

Reply:
I fixed this. Again, thanks @wa-Nadoo and @ronag

Reply:
Sorry, I need to reopen this. I don't think that the behavior was as you explained in the previous implementation. In fact, in `http_server` and `_http_client`,  the points where an error is emitted are in the abort logic, and only if there are listeners attached to the event. That, and the fact the IncomingMessage wasn't implementing a standard destroy (therefore, it wasn't itself emitting an `error` event), made it emit an error event only in case of abort if there were event handlers.

Something that I think proves this is the fact that in `_http_client`, the `error` event of `res` is no bubbled up to `req`. The same goes for `_http_server`. I think maybe this example explains this: https://github.com/nodejs/node/pull/33035#discussion_r543109670.`

I added some tests, ptal.